CypherpunkArmory/holepunch
# [337] Change Account Password

## What changes does this PR introduce?
Adds an endpoint for changing the user password

## Where should the reviewer start?
`app/routes/Authentication.py#99` is where the new route is defined.  The template for the email to be sent is defined in `app/jobs/email.py#48`

## Has this been manually tested? How?
Hit the endpoints via Postman:

1. Registered new user.
2. Logged in with credentials.
3. Reset password with the same password (received expected failure).
4. Reset password with a new password successfully.
5. Log in with old password (received expected failure).
6. Logged in with new password successfully.

## What GIF best describes this PR or how it makes you feel?
![i'm in the codebase now](https://media.giphy.com/media/DFRghyjqRDkJO/giphy.gif)
